pkg/nrf_modem/contrib/nrf_modem_hal.c

```diff
@@ -0,0 +1,106 @@
+/*
```

Comment:
Because the libmodem.a call these functions. Unfortunately, it seems  that this library relies (without any mention in the doc as far as I can tell) on nordic sdk for using the IPC peripheral which is needed for proper operations. I had to reimplement these functions using Nordic terminology so I could successfully link the lib against my code as I don't want to pull their whole SDK into RIOT...

Reply:
I see. So you took the implementation from [this library](https://github.com/NordicSemiconductor/nrfx/blob/v2.7.0/drivers/src/nrfx_ipc.c)? Wouldn't it be cheaper to add it as a package? I think it may turn to be an overhead to maintain a driver that is basically what they already have upstream. Or is there maybe some advantage other than not pulling the package once, which I fail to see here?

Reply:
> So you took the implementation from [this library](https://github.com/NordicSemiconductor/nrfx/blob/v2.7.0/drivers/src/nrfx_ipc.c)?

Yes.

> Wouldn't it be cheaper to add it as a package? 

I don't think so. This can be done but at first I wanted to avoid all the pain adding such a big pkg for a single file in one PR.